cartoframes/assets/templates/vector/scripts/map.js.j2

```diff
@@ -49,11 +52,17 @@ function onReady() {
     const viz = new carto.Viz(elem['viz']);
 
     const layer = new carto.Layer(
-        'layer' + idx,
-        temp_source,
-        viz
+      'layer' + idx,
+      temp_source,
+      viz
     );
 
+    try {
+      layer._updateLayer.catch(displayError);
```

Comment:
Hmm, I don't fully understand this, and seems kind of internal. Was this left on purpose?

Reply:
We've discussed this in the past: the main issue is that we need to refactor, at some point, the Layer CARTO VL class because the constructor is asynchronous. _updateLayer is a Promise created in the Layer constructor, and we need to catch it to avoid the `Uncaught in promise` error and be able to show it.